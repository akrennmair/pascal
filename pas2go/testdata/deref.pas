{$mode ISO}
program test;

type y = record
		c : ^integer;
	end;

var x : record
		a : ^integer;
		b : ^y;
	end;

begin
	x.a^ := x.b^.c^;
	x.b^.c^ := 23;
	x.b^.c^ := x.a^
end.
