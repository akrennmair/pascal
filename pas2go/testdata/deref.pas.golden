package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		y struct {
			c *int
		}
	)

	var (
		x struct {
			a *int
			b *struct {
				c *int
			}
		}
	)
	_ = x

	(*x.a) = (*(*x.b).c)
	(*(*x.b).c) = 23
	(*(*x.b).c) = (*x.a)
}
